src/simulator_core/entities/assets/ates_cluster.py

```diff
@@ -0,0 +1,276 @@
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""atesCluster class."""
+from typing import Dict
+import os
+import math
+
+from simulator_core.entities.assets.asset_abstract import AssetAbstract
+from simulator_core.entities.assets.asset_defaults import (
+    DEFAULT_TEMPERATURE,
+    DEFAULT_TEMPERATURE_DIFFERENCE,
+    PROPERTY_HEAT_DEMAND,
+    PROPERTY_MASSFLOW,
+    PROPERTY_PRESSURE_RETURN,
+    PROPERTY_PRESSURE_SUPPLY,
+    PROPERTY_TEMPERATURE_RETURN,
+    PROPERTY_TEMPERATURE_SUPPLY,
+)
+
+from simulator_core.entities.assets.esdl_asset_object import EsdlAssetObject
+from simulator_core.solver.network.assets.production_asset import ProductionAsset
+from simulator_core.entities.assets.utils import (
+    heat_demand_and_temperature_to_mass_flow,
+)
+
+path = os.path.dirname(__file__)
+import jnius_config  # noqa
+
+jnius_config.add_classpath(os.path.join(path, 'bin/jfxrt.jar'))
+jnius_config.add_classpath(os.path.join(path, 'bin/rosim-batch-0.4.2.jar'))
+from jnius import autoclass  # noqa
+
+javaioFile = autoclass('java.io.File')
+RosimSequential = autoclass('tno.calc.RosimSequential')
+
+
+class AtesCluster(AssetAbstract):
+    """A AtesCluster represents an asset that consumes heat and produces heat."""
+
+    temperature_supply: float
+    """The supply temperature of the asset [K]."""
+
+    temperature_return: float
+    """The return temperature of the asset [K]."""
+
+    thermal_power_allocation: float
+    """The thermal for injection (positive) or production (negative) by the asset [W]."""
+
+    mass_flowrate: float
+    """The flow rate going in or out by the asset [kg/s]."""
+
+    def __init__(
+            self,
+            asset_name: str,
+            asset_id: str,
+    ):
+        """Initialize a AtesCluster object.
+
+        :param str asset_name: The name of the asset.
+        :param str asset_id: The unique identifier of the asset.
+        """
+        super().__init__(asset_name=asset_name, asset_id=asset_id)
+
+        self.temperature_supply = DEFAULT_TEMPERATURE
+        self.temperature_return: float = DEFAULT_TEMPERATURE - DEFAULT_TEMPERATURE_DIFFERENCE
+        self.thermal_power_allocation = 0
+        self.mass_flowrate = 0
+        self.solver_asset = ProductionAsset(name=self.name, _id=self.asset_id)
+        # ATES default properties
+        self.aquifer_depth = 300.0
+        self.aquifer_thickness = 45.0
+        self.aquifer_mid_temperature = 17.0
+        self.aquifer_net_to_gross = 1.0
+        self.aquifer_porosity = 0.3
+        self.aquifer_permeability = 10000.0
+        self.aquifer_anisotropy = 4.0
+        self.salinity = 10000.0
```

Comment:
What is the unit, this seems a very high value? Can you add units as comment to all?

Reply:
unit has been added